com.unity.ml-agents.extensions/Runtime/Sensors/GridSensorComponent.cs

```diff
@@ -202,68 +184,149 @@ public int ObservationStacks
             set { m_ObservationStacks = value; }
         }
 
+        [HideInInspector, SerializeField]
+        internal bool m_UseOneHotTag = true;
+        /// <summary>
+        /// Whether to use one-hot representation of detected tag as observation.
+        /// Note that changing this after the sensor is created has no effect.
+        /// </summary>
+        public bool UseOneHotTag
+        {
+            get { return m_UseOneHotTag; }
+            set { m_UseOneHotTag = value; }
+        }
+
+        [HideInInspector, SerializeField]
+        internal bool m_CountColliders = false;
+        /// <summary>
+        /// Whether to use the number of count for each detectable tag as observation.
+        /// Note that changing this after the sensor is created has no effect.
+        /// </summary>
+        public bool CountColliders
+        {
+            get { return m_CountColliders; }
+            set { m_CountColliders = value; }
+        }
+
         /// <inheritdoc/>
         public override ISensor[] CreateSensors()
         {
-            m_Sensor = new GridSensor(
-                m_SensorName,
+            List<ISensor> m_Sensors = new List<ISensor>();
+            m_BoxOverlapChecker = new BoxOverlapChecker(
                 m_CellScale,
                 m_GridSize,
                 m_RotateWithAgent,
-                m_ChannelDepths,
-                m_DetectableObjects,
                 m_ColliderMask,
-                m_DepthType,
                 gameObject,
-                m_CompressionType,
-                m_MaxColliderBufferSize,
-                m_InitialColliderBufferSize
+                m_DetectableTags,
+                m_InitialColliderBufferSize,
+                m_MaxColliderBufferSize
             );
 
-            if (ObservationStacks != 1)
+            // debug data is positive int value and will trigger data validation exception if SensorCompressionType is not None.
+            m_DebugSensor = new GridSensorBase("DebugGridSensor", m_CellScale, m_GridSize, m_DetectableTags, SensorCompressionType.None);
+            m_BoxOverlapChecker.RegisterDebugSensor(m_DebugSensor);
+
+            if (m_UseOneHotTag)
```

Comment:
Move the creation of OneHotGridSensor and CountingGridSensor to the default impl of GetGridSensors?

Reply:
I can see how this make sense, but then when GetGridSensors is overridden the bool `m_UseOneHotTag` specified in UI will have no effect which is a bit confusing.

Reply:
If you override GetGridSensors(), you can still call base.GetGridSensors() (or not)

Reply:
ok, I moved it.